src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/InstantiateArgumentExceptionsCorrectly.cs

```diff
@@ -234,10 +235,20 @@ private static bool MatchesParameterCore(ISymbol? symbol, string stringArgumentV
         {
             foreach (IParameterSymbol parameter in symbol.GetParameters())
             {
+                // If the parameter name matches exactly, it's a match.
                 if (parameter.Name == stringArgumentValue)
                 {
                     return true;
                 }
+
+                // If the string argument begins with the parameter name followed by punctuation, it's also considered a match.
+                // e.g. "arg.Length", "arg[0]", etc.
+                if (stringArgumentValue.Length > parameter.Name.Length &&
+                    stringArgumentValue.StartsWith(parameter.Name, StringComparison.Ordinal) &&
+                    char.IsPunctuation(stringArgumentValue, parameter.Name.Length))
```

Comment:
LGTM

There are some potentially other interesting scenarios like `-name` or `name + length`, but I think we can look at those later if we find actual scenarios.

Reply:
FWIW, if memory serves, the guidance had been that the parameter should just be the parameter name and then the message should include the specifics of what about it was wrong, e.g. mentioning its Length or whatever.  That's just not feasible if someone wants to use the public throw helpers.